### PR TITLE
Including body and title of GitHub issues.

### PR DESCRIPTION
**Pull request by tobie**
Adresses some but not all of the issues brought up in #241.

Needs tests, but those will need to rely on a mock server for requests.

Also needs doc.

Help welcomed. :)

### CODE REVIEW

File: js/github.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Helpers for the GitHub API.
+
+define(
+    [],
+    function () {
+        function findNext(header) {
+            var m = (header||"").match(/<([^>]+)>\s*;\s*rel="next"/);
```
Comment (marcoscaceres) [inline js/github.js]: A comment describing the regex would be nice.

File: js/github.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            return (m && m[1]) || null;
+        }
+        
+        function fetch(url) {
+            return $.ajax(url);
```
Comment (marcoscaceres) [inline js/github.js]: why not feature test here for fetch API? :) Could be neat, as it's basically the same API as jQuery.

File: js/core/issues-notes.js [hunk 3]
```diff
@@ -67,20 +72,47 @@ define(
...
                                 $inno.removeAttr("title");
                             }
                             $div.append($tit);
                             $inno.replaceWith($div);
-                            $div.append($inno.removeClass(report.type).removeAttr('data-number'));
+                            var body = $inno.removeClass(report.type).removeAttr('data-number');
+                            if (ghIssue && !body.text().trim()) {
+                                body = marked(ghIssue.body, {
+                                    gfm: true,
+                                    pedantic: false,
```
Comment (marcoscaceres) [inline js/core/issues-notes.js]: Maybe describe what these do?

File: js/github.js (hunk 2 above)
Comment (tobie) [inline js/github.js]: The return promise isn't fulfilled with the same thing (JSON here, a `Request` object for `fetch`).

Comment (marcoscaceres) [inline js/github.js]: ah, yeah... nm.

File: js/core/issues-notes.js (hunk 3 above)
Comment (tobie) [inline js/core/issues-notes.js]: Dude, I actively practice cargo-culting. They're supposed to do wonderful things. In practice, I'm not so sure.

File: js/core/issues-notes.js [hunk 4]
```diff
@@ -67,20 +72,47 @@ define(
...
+                if ($ins.length) {
+                    if (conf.githubAPI) {
+                        github.fetch(conf.githubAPI).then(function(json) {
+                            return github.fetchIndex(json.issues_url);
+                        }).then(function (issues) {
+                            issues.forEach(function(issue) {
+                                ghIssues[issue.number] = issue;
+                            });
+                            handleIssues($ins);
+                            onEnd();
```
Comment (marcoscaceres) [inline js/core/issues-notes.js]: nit: can't the `onEnd();` go at the end? These ifs all end with it?

File: js/core/issues-notes.js (hunk 3 above)
Comment (marcoscaceres) [inline js/core/issues-notes.js]: heh :dancer:

File: js/core/issues-notes.js (hunk 4 above)
Comment (marcoscaceres) [inline js/core/issues-notes.js]: Sorry, I'm stupid. This is deferred.

**Comment by marcoscaceres** [comment]
This looks great, @tobie ! :D

File: js/core/issues-notes.js (hunk 4 above)
Comment (tobie) [inline js/core/issues-notes.js]: Nope, some of the code is async, so it would get called too early.

**Comment by marcoscaceres** [comment]
Mind posting a screenshot of the output?

**Comment by tobie** [comment]
gh-issues

**Comment by marcoscaceres** [comment]
I think I just cried tears of joy

File: js/core/issues-notes.js (hunk 3 above)
Comment (darobin) [inline js/core/issues-notes.js]: Is applying marked here sufficient to process the content? See https://twitter.com/domenic/status/604507705223593984

Comment (tobie) [inline js/core/issues-notes.js]: I wasn't aware there was a way to get at the HTML body directly. Can't find it documented anywhere.

Comment (tobie) [inline js/core/issues-notes.js]: Found it. Will give it a spin. Awesome if it works.

**Comment by tobie** [comment]
OK, so using Media Types to get `body_html` works and is all kinds of cool. Here's a pic to get @marcoscaceres to weep some more.

gh-issues

Thanks @darobin and @domenic for the pointer.

**Comment by domenic** [comment]
This is amazing. @tabatkins Bikeshed needs to steal this ;)

**Comment by marcoscaceres** [comment]
YOU STOP COPYING OUR FEATURES - PATENT PENDING! :)

**Comment by tabatkins** [comment]
TOO LATE I'M ALREADY WRITING CODE

**Comment by marcoscaceres** [comment]
WE WILL SEE YOU IN COURT! 
screenshot 2015-06-01 16 42 52

**Comment by tabatkins** [comment]
As you may  have noticed, NOBODY CAN STOP ME.

**Comment by darobin** [comment]
This LGTM, any reason not to merge? I mean other than obligate whining about plagiarism?

**Comment by domenic** [comment]
Breaking news: Bikeshed implements inline GH issues first, respec stuck in committee. ;)

**Comment by halindrome** [comment]
Ship it!

On Tue, Jun 2, 2015 at 8:34 AM, Domenic Denicola <notifications@github.com>
wrote:

> Breaking news: Bikeshed implements inline GH issues first, respec stuck in
> committee. ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/w3c/respec/pull/443#issuecomment-107954487>.
>



-- 
Shane McCarron
halindrome@gmail.com

**Pull request merged**